File: problems/3.py
# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143 ?

def solution(n: int = 600851475143) -> int:
    i = 2
    ans = 0
    if n == 2:
        return 2
    while n > 2:
        while n % i != 0:
            i += 1
        ans = i
        while n % i == 0:
            n = n // i
        i += 1
    return int(ans)


if __name__ == "__main__":
    print(f"{solution() = }")

File: problems/c/3.c
// The prime factors of 13195 are 5, 7, 13 and 29.
// What is the largest prime factor of the number 600851475143 ?

#include <stdio.h>

int main() {
    long long int n = 600851475143;
    long long int i = 2;
    while (i * i < n) {
        while (n % i == 0) {
            n /= i;
        }
        i++;
    }
    printf("%lld", n);
}